logs_api/lib/opentelemetry/logs/logger_provider.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Logs
+    # No-op implementation of a logger provider
+    class LoggerProvider
+      # Returns a {Logger} instance.
+      #
+      # @param name [optional String] Instrumentation package name
+      # @param version [optional String] Instrumentation package version
+      # @param schema_url [optional String] Schema URL recorded in the emitted
+      #   telemetry.
+      # @param attributes [optional Hash] Specifies the instrumentation scope
+      #   attributes to associate with emitted telemetry.
+      #
+      # @return [OpenTelemetry::Logs::Logger]
+      #
+      # @api public
+      def logger(name: nil, version: nil, schema_url: nil, attributes: {})
```

Comment:
The [spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/noop.md#logger-creation) allows us to return a single logger instance for the Noop implementation. It says:

> Since all Loggers will hold the same empty state, a LoggerProvider MAY return the same Logger instances to all creation requests.

We might be able to get away with defining a `NOOP_LOGGER = Logger.new` and always returning it. 

Not to complicate matters, but for tracing, it was expected that if an application started with the NoOp implementation, that tracers would become operational if an SDK was installed later. You can see the API installs a `ProxyTracerProvider` (which hands out `ProxyTracer`s). The delegate for the proxy tracers are initially API tracers, but they can be upgraded to SDK tracers.

I was trying to figure out if the same requirement exists for Logging, but I wasn't immediately able to track that down. We should find that out as that will impact what we want to do here.

If we do keep this as is, we can drop the `OpenTelemery::Logs::` namespacing.

Reply:
Great idea. I'll do a bit of digging to find out whether the same requirement exists for Logs.

I've incorporated `NOOP_LOGGER` in dc966f0a153a386f91614742c3c6264422b9e6ca